eventbus.go

```diff
@@ -35,8 +35,18 @@ func (e EventBus) UnSubscribe(eventName string, ch DataChannel) {
 
 	for idx, subscriber := range subscribers {
 		if subscriber == ch {
-			e.events[eventName] = append(subscribers[:idx], subscribers[idx+1:]...)
-			return
+			// the order of subscribers does not matter
+			subscribers[idx], subscribers[len(subscribers)-1] = subscribers[len(subscribers)-1], nil
+			e.events[eventName] = subscribers[:len(subscribers)-1]
+
+			// drain the channel
+			for {
+				select {
+				case <-ch:
+				default:
+					return
+				}
+			}
```

Comment:
close(ch)
return

Reply:
It would be better if we don't close it based on [The Channel Closing Principle](https://go101.org/article/channel-closing.html)

Reply:
@arberiii "don't close a channel from the receiver side and don't close a channel if the channel has multiple concurrent senders"
It seems that we are meeting both criterias:
1. Eventbus is the sender for the channel
1. Eventbus is the only sender for the channel

Please let me know if you disagree.

Reply:
I do agree!